### PR TITLE
feat: [CI-11723]: Introduced RUN_ON_HOST_ONLY flag

### PR DESCRIPTION
**Pull request by devkimittal**

### CODE REVIEW

**Review by HenAmar — APPROVED** [review]

File: handler/setup.go [hunk 1]
```diff
@@ -48,8 +48,13 @@ func HandleSetup(config *config.Config) http.HandlerFunc {
 			return
 		}
 		id := s.ID
+		if config.Runner.RunOnHostOnly {
+			mountDockerSocket := false
+			s.MountDockerSocket = &mountDockerSocket
+		}
 
 		updateVolumes(s)
+		s.Envs["WORKSPACE"] = getWorkSpacePath(s)
```
Comment (vistaarjuneja) [inline handler/setup.go:57]: Let's set all these from Java and avoid adding custom code to the runner.
Currently we are in the middle of an effort to consolidate a lot of these runners to a single repo.
The goal is for the docker runner to not have any custom logic and just get task requests and execute them accordingly.

File: handler/step.go [hunk 2]
```diff
@@ -38,6 +38,11 @@ func HandleStartStep(config *config.Config) http.HandlerFunc {
 			return
 		}
 
+		if config.Runner.RunOnHostOnly {
```
Comment (vistaarjuneja) [inline handler/step.go:41]: All this should get set from the platform

File: engine/docker/docker.go [hunk 3]
```diff
@@ -208,8 +208,10 @@ func (e *Docker) Destroy(ctx context.Context, pipelineConfig *spec.PipelineConfi
 	}
 
 	// cleanup the network
-	if err := e.client.NetworkRemove(ctx, pipelineConfig.Network.ID); err != nil {
-		logrus.WithField("network", pipelineConfig.Network.ID).WithField("error", err).Warnln("failed to remove network")
+	if e.IsDockerEnabled(pipelineConfig) {
```
Comment (vistaarjuneja) [inline engine/docker/docker.go:211]: Don't do this here. Don't invoke a `docker.Destroy` if docker is not installed.

File: engine/docker/docker.go [hunk 4]
```diff
@@ -223,6 +225,13 @@ func (e *Docker) Destroy(ctx context.Context, pipelineConfig *spec.PipelineConfi
 	return nil
 }
 
+func (e *Docker) IsDockerEnabled(pipelineConfig *spec.PipelineConfig) bool {
```
Comment (vistaarjuneja) [inline engine/docker/docker.go:228]: This doesn't belong here. This layer just does docker operations - it doesn't check whether docker is installed or not. That should be whoever invokes this code.

**Review by vistaarjuneja — CHANGES_REQUESTED** [review]

File: handler/step.go (hunk 2 above)
Comment (devkimittal) [reply handler/step.go:41]: platform meaning ci manager?

**Comment by vistaarjuneja** [comment]
Can we just print a warning message if docker is not installed instead of adding separate config to do this?
Someone could be running the docker runner and simply not need docker as all steps run on the host. So docker shouldn't be a hard requirement. If their pipeline fails, they should be able to see in the error messages that it was because of docker not being installed. As such, this is a "local runner" more than a "docker runner"

**Pull request closed without merging**